### PR TITLE
feat(regional): Italian Localization

### PR DESCRIPTION
**Pull request by gaurav-naik**
Added setup.py for italy under regional
Included fiscal regimes, tax exemption reasons and custom fields

Issue #16259

@umairsy @cosimot @mascor

### CODE REVIEW

File: erpnext/regional/report/electronic_invoice_register/electronic_invoice_register.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ "modified": "2019-01-13 19:03:56.187748", 
+ "modified_by": "Administrator", 
+ "module": "Regional", 
+ "name": "Electronic Invoice Register", 
+ "owner": "Administrator", 
+ "prepared_report": 0, 
+ "ref_doctype": "Sales Invoice", 
+ "report_name": "Electronic Invoice Register", 
+ "report_type": "Script Report", 
+ "roles": []
```
Comment (nabinhait) [inline erpnext/regional/report/electronic_invoice_register/electronic_invoice_register.json]: You should add permissions for this report via `italy/setup.py`.

File: erpnext/regional/italy/utils.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+	if filters.get("from_date"): conditions["posting_date"] = (">=", filters["from_date"])
+	if filters.get("to_date"): conditions["posting_date"] = ("<=", filters["to_date"])
+	
+	if filters.get("from_date") and filters.get("to_date"):
+		conditions["posting_date"] = ("between", [filters.get("from_date"), filters.get("to_date")])
+	
+	return conditions
+
+def download_zip(files, output_filename):
```
Comment (nabinhait) [inline erpnext/regional/italy/utils.py]: This function can be reused and moved to frappe

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/regional/italy/utils.py (hunk 2 above)
Comment (gaurav-naik) [reply erpnext/regional/italy/utils.py]: https://github.com/frappe/frappe/pull/6853

File: erpnext/regional/report/electronic_invoice_register/electronic_invoice_register.json (hunk 1 above)
Comment (gaurav-naik) [reply erpnext/regional/report/electronic_invoice_register/electronic_invoice_register.json]: Shall I "git rebase hotfix" locally?

**Comment by gaurav-naik** [comment]
Closing on account of merging issues. Follow https://github.com/frappe/erpnext/pull/16535

**Pull request closed without merging**